crypto/mem_sec.c

```diff
@@ -23,6 +23,20 @@
 #ifndef OPENSSL_NO_SECURE_MEMORY
 # if defined(_WIN32)
 #  include <windows.h>
+#  if defined(WINAPI_FAMILY_PARTITION) \
+     && !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
+/*
+ * While VirtualLock is available under the app partition (e.g. UWP),
+ * the headers do not define the API. Define it ourselves instead.
+ */
+WINBASEAPI
+BOOL
+WINAPI
+VirtualLock(
```

Comment:
More details [here](https://github.com/openssl/openssl/pull/18265#issuecomment-1121389225) and [here](https://github.com/openssl/openssl/pull/18265#issuecomment-1121601427).

Essentially, while VirtualLock is available in the import library (and is not blocked under UWP, it would be missing from the import lib if it was), there is a bug in both the documentation and header files. The headers do not forward declare the API for UWP code, and the docs say "desktop apps only" instead of "desktop apps | UWP apps". So to be able to use the API, we have to manually forward declare it until these bugs are resolved.

Reply:
`windows.h` pulls in `memoryapi.h` by default already, so manually including it does not resolve the issue.